lib/chef/provisioning/aws_driver/aws_resource.rb

```diff
@@ -133,13 +133,13 @@ def self.aws_sdk_type(sdk_class,
                         load_provider: true,
                         id: :name,
                         aws_id_prefix: nil)
-    self.resource_name = self.dsl_name
+    self.resource_name = convert_to_snake_case(self.name.split('::')[-1])
     @aws_sdk_class = sdk_class
     @aws_sdk_class_id = id
     @aws_id_prefix = aws_id_prefix
 
     # Go ahead and require the provider since we're here anyway ...
-    require "chef/provider/#{resource_name}" if load_provider
+    require "chef/provider/#{resource_name.to_s}" if load_provider
```

Comment:
This should be unnecessary, .to_s is automatic when you embed something in a string.